sedbackend/apps/cvs/vcs/storage.py

```diff
@@ -918,31 +918,28 @@ def edit_vcs_table(db_connection: PooledMySQLConnection, project_id: int, vcs_id
 
 def remove_duplicate_names(project_id: int, vcs_id: int, rows: List[models.VcsRowPost]):
     for i in range(0, len(rows)):
+        dups: List[Tuple[int, models.VCSSubprocessPost]] = []
         for j in range(i + 1, len(rows)):
             if rows[i].iso_process is None:
                 break
-            dups: List[Tuple[int, models.VCSSubprocessPost]] = []
-            if rows[i].iso_process == rows[j].iso_process:
+            if rows[i].iso_process and rows[i].iso_process == rows[j].iso_process:
                 process = vcs_impl.get_iso_process(rows[i].iso_process)
-                sub = models.VCSSubprocessPost(name=process.name + str(len(dups) + 2), parent_process_id=process.id)
+                sub = models.VCSSubprocessPost(name=process.name + " (" + str(len(dups) + 1) + ")",
+                                               parent_process_id=process.id)
                 dups.append((j, sub))
 
-            if len(dups) > 0:
-                subfst = models.VCSSubprocessPost(name=process.name + "1", parent_process_id=process.id)
-                dups.append((i, subfst))
-
-                for index, dup in dups:
-                    subp = vcs_impl.create_subprocess(project_id, vcs_id, dup)
-                    rowPost = models.VcsRowPost(
-                        id=rows[index].id,
-                        index=rows[index].index,
-                        stakeholder=rows[index].stakeholder,
-                        stakeholder_needs=rows[index].stakeholder_needs,
-                        stakeholder_expectations=rows[index].stakeholder_expectations,
-                        iso_process=None,
-                        subprocess=subp.id)
-
-                    rows[index] = rowPost
+        for index, dup in dups:
+            subp = vcs_impl.create_subprocess(project_id, vcs_id, dup)
```

Comment:
Also check if there are other storage-methods that calls impl-methods as this should be avoided as much as possible to prevent the server from running out of db-connections (that is one of the purposes of splitting it into impl and storage).

Reply:
I removed all the uses of impl functions and I did an attempt in a batch insert but I need to get the ids of the newly inserted rows and I could not figure out a way to get the ids from each row because insert_last_id() only returns the first row of the batch in MySQL.

Reply:
I think this explains how it can be done pretty well:

https://stackoverflow.com/questions/7333524/how-can-i-insert-many-rows-into-a-mysql-table-and-return-the-new-ids

Reply:
Thank you it worked! I read that solution before but I was unsure of how reliant it is but it looks like it works